docs/decisions/00XX-workflow_composability_design.md

```diff
@@ -0,0 +1,245 @@
+  - Input type compatibility is checked using existing is_type_compatible between parent edge source and sub-workflow start executor types; adapter output types are validated before send_message.
+- Example:
+
+```python
+analysis = (
+    ConcurrentBuilder()
+    .participants([operation, compliance])
+    .build()
+    .as_executor(id="analysis", adapter=pick_latest_assistant)
+)
+
+workflow = (
+    WorkflowBuilder()
+    .add_edge(orchestrator, analysis)
+    .add_edge(analysis, aggregator_executor)
+    .set_start_executor(orchestrator)
+    .build()
+)
+```
+- Pros: minimal code surface; reuses WorkflowExecutor and existing validation. Cons: maintains nested execution boundary (double superstep scheduling, separate checkpoint lineage), and post-build graph edits are still indirect.
+
+## Option 2: Inline Fragments With Builder Merge (connect-first API)
+- Keep fragment inlining but make connection verbs explicit and fluent via `.connect(...)`.
+- `WorkflowConnection`:
+  - `builder`: WorkflowBuilder holding the fragment wiring.
+  - `entry`: canonical entry executor id.
+  - `exits`: executor ids that are safe to target (default terminal outputs or last adapters).
+  - `contract`: input/output type sets plus semantics tags for validation (reuses Option 3 contracts if available).
+- Production:
+  - High-level builders expose `.as_connection()` returning connection metadata without building a Workflow.
+  - Built workflows expose `.as_connection()` by cloning topology into a new builder (immutable source).
+- Composition API (renamed for clarity):
+  - Single verb: `connect(...)` handles both merge and wiring. No `add_fragment`.
+  - Accepted inputs:
+    - `connect(fragment_or_workflow, *, prefix=None, source=None, target=None, adapter=None)`
+    - `connect(source_executor_id_or_port, target_executor_id_or_port, *, adapter=None)`
+  - If `fragment_or_workflow` is provided:
+    - Merge its builder into the caller with optional `prefix` to avoid ID collisions.
+    - Return a handle with `.start` and `.outputs` to allow chaining.
+    - If `source` is provided, wire `source -> fragment.start`.
+    - If `target` is provided, wire `fragment.outputs[0] -> target` (or all outputs if specified).
+  - `FragmentHandle.start` alias for entry; `FragmentHandle.outputs` alias for exits to keep names concise in chaining.
+  - Optional chaining: `builder.connect(orchestrator, analysis.start).connect(analysis.outputs[0], aggregator)`.
+- Naming shift: prefer `connect` over `add_edge` for user-facing fluent APIs; keep `add_edge` under the hood for compatibility.
+- Type safety:
+  - `connect` enforces compatibility between source output types and target input types (or fragment contract).
+  - Allow optional `adapter` param to inject a converter executor inline if strict types differ (compatible with Option 3 registry).
+- Example:
+
+```python
+analysis = (
+    ConcurrentBuilder()
+    .participants([operation, compliance])
+    .as_connection()
```

Comment:
is this even needed? couldn't the next step just accept a `unbuild` workflowbuilder of any type?